yt_dlp/extractor/locipo.py

```diff
@@ -0,0 +1,149 @@
+
+from .common import InfoExtractor
+from ..utils import (
+    clean_html,
+    int_or_none,
+    parse_iso8601,
+    parse_qs,
+    str_or_none,
+    url_or_none,
+)
+from ..utils.traversal import require, traverse_obj
+
+
+class LocipoBaseIE(InfoExtractor):
+    _BASE_URL = 'https://locipo.jp'
+
+    def _call_api(self, path, item_id):
+        return self._download_json(
+            f'https://api.locipo.jp/api/v1/{path}', item_id)
+
+
+class LocipoIE(LocipoBaseIE):
+    _VALID_URL = r'https?://locipo\.jp/(?P<type>creative|embed)/(?P<id>[^/?#]+)?'
```

Comment:
sorry, type was unnecessary
also changing to a more accurate regex
```suggestion
    _VALID_URL = r'https?://locipo\.jp/(?:creative|embed)/(?:\?(?:[^#]+&)?id=)?(?P<id>[\da-f]{8}(?:-[\da-f]{4}){3}-[\da-f]{12})'
```